### PR TITLE
Simplification of ZUML for Southern Ocean

### PR DESCRIPTION
**Pull request by arnaldorusso**
Travis is faling on [`woa_subset`](https://travis-ci.org/arnaldorusso/python-oceans/jobs/53881932#L532).
Can I help in something?

Cheers,
Arnaldo.

### CODE REVIEW

**Comment by ocefpaf** [comment]
Can you do a few things?

- Consider consolidating them into one with keywords for each case (Not sure if it is the best option, just think about your user case and see if it is best).
- Add an entry on the `CHANGES.txt` file
- Add a docstring example in like:

```python
Examples
--------
>>> zuml_so(MOL)
42
```

Docstrigs like that will be found by `py.test`, so we get the testing for free.  Also, they are examples on how to use the function.

PS: Regarding `woa_subset` see #13.

**Comment by arnaldorusso** [comment]
Hi @ocefpaf 
I prefer to maintain in separated ways on how to calculate Zuml.
The python transcription of [Argo Mixed Layers](http://mixedlayer.ucsd.edu/) was only applied for the first method, since other way could be implemented inside the same function using keywords.

I have also used the sphinx suggestion on how to include math expression inline.

**Comment by arnaldorusso** [comment]
Regarding `CHANGES.txt`, the entries should be included in the last version `Version 0.2.3, 23-Jan-2015.` or I have to rename a new Version as `Version 0.2.4`?

File: oceans/sw_extras/sw_extras.py [hunk 1]
```diff
@@ -855,11 +855,39 @@ def zmld_so(s, t, p, threshold=0.05, smooth=None):
...
+        27.,  28.,  29.,  30.,  31.,  32.,  33.,  34.])
+    >>> swe.zuml_so(s, t, p)
+    20.0
+
     References
     ----------
-    Mitchell B. G., Holm-Hansen, O., 1991. Observations of modeling of the
-        Antartic phytoplankton crop in relation to mixing depth. Deep Sea
-        Research, 38(89):981-1007. doi:10.1016/0198-0149(91)90093-U
+    .. [1] Mitchell B. G., Holm-Hansen, O., 1991. Observations of modeling of the
```
Comment (ocefpaf) [inline oceans/sw_extras/sw_extras.py]: Great!  We should start the docs engine soon.

File: oceans/sw_extras/sw_extras.py [hunk 2]
```diff
@@ -890,6 +919,37 @@ def zmld_boyer(s, t, p):
...
+        2.1395,  2.1407,  2.143 ,  2.1416,  2.1166,  2.0937,  2.1195,
+        2.1355,  2.0627,  1.6204,  1.3384,  1.1135,  0.9938,  0.9379,
+        0.9064,  0.8811,  0.8548,  0.8407,  0.8235,  0.7893,  0.779 ,
+        0.782 ,  0.7765])
+    >>> p = np.array([
+        5.,   6.,   7.,   8.,   9.,  10.,  11.,  12.,  13.,  14.,  15.,
+        16.,  17.,  18.,  19.,  20.,  21.,  22.,  23.,  24.,  25.,  26.,
+        27.,  28.,  29.,  30.,  31.,  32.,  33.,  34.])
+    >>> swe.zuml_boyer(s, t, p)
+    (20.0, 20.0)
```
Comment (ocefpaf) [inline oceans/sw_extras/sw_extras.py]: Nice!  Just for the "oceanographic curiosity" sake: do you want to go the extra mile and add an example where these values differ?

**Comment by ocefpaf** [comment]
> Regarding CHANGES.txt, the entries should be included in the last version Version 0.2.3, 23-Jan-2015. or I have to rename a new Version as Version 0.2.4?

Add a new entry for the future version 0.2.4

File: oceans/sw_extras/sw_extras.py [hunk 3]
```diff
@@ -855,11 +855,43 @@ def zmld_so(s, t, p, threshold=0.05, smooth=None):
...
+    ------
+    zuml : int
+        Depth of the upper mixed layer
+
+    Examples
+    --------
+    >>> from oceans import sw_extras as swe
+    >>> import numpy as np
+    >>> s = np.array([
+        34.0991,  34.0978,  34.0978,  34.1006,  34.1023,  34.1041,
```
Comment (ocefpaf) [inline oceans/sw_extras/sw_extras.py]: As you can see from the travis failure, this syntax is not allowed in docstrings.  See [this](https://github.com/pyoceans/python-oceans/blob/master/oceans/sw_extras/sw_extras.py#L250---251) for an example on how to do line continuations on docstrings.

BTW: you can run the tests locally, before send the PR with the command `py.test` issue from the package root directory (where `setup.py` is).

**Comment by arnaldorusso** [comment]
Travis continues to FAIL while reading [doctest](https://travis-ci.org/pyoceans/python-oceans/jobs/53948801#L572).

File: CHANGES.txt [hunk 4]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ---------
 
+Version 0.2.4, 11-Mar-2015.
+
+* Updating Mixed Layer Depth functions
+* Examples included
```
Comment (ocefpaf) [inline CHANGES.txt]: No need for this line, in theory examples a given and not a feature that needs announcing :grimacing:

**Pull request closed without merging**